ui/src/components/Layout.tsx

```diff
@@ -30,6 +32,13 @@ type LayoutProps = {
 };
 const Layout: React.FC<LayoutProps> = function (props) {
   const { children } = props;
+  const auth0Context = React.useContext(Auth0Context);
+  const profileContext = useProfile();
+  if (!profileContext?.profile) {
```

Comment:
is not doing anything the expected behavior if these contexts are missing?

Reply:
Kinda overlooked this. I've just updated useProfile to throw Error if it's not used in ProfileProvider.

DIdn't touch Auth0Context for now cos I think it could be refactored abit.